pkg/controller/machineset/controller.go

```diff
@@ -188,12 +189,13 @@ func (r *ReconcileMachineSet) Reconcile(ctx context.Context, request reconcile.R
 				"The AuthoritativeAPI is set to %s", string(machineSet.Status.AuthoritativeAPI),
 			))
 
-			machineSet, err := updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
+			_, err := updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
 			if err != nil {
-				klog.Errorf("%v: error updating status: %v", machineSetCopy.Name, err)
+				klog.Errorf("%v: error updating status: %v", machineSetName, err)
+				return reconcile.Result{}, fmt.Errorf("error updating status: %w", err)
```

Comment:
I'm not sure we really need to error here, given it's only a status update error where erroring or not produce the same behaviour (apart from the re-reconcile).

However, for UX in the mapicapi migration I think it's worth retrying.